### PR TITLE
Samza-2540: Legacy Task application are missing app.run.id for both batch & stream jobs

### PR DESCRIPTION
**Pull request by Sanil15**
**Symptom [Bug fix]:** 
- LegacyTaskApplication (ones that do not use descriptors) are missing app.run.id config which makes container placements nonusable with them
- Ideally app.run.id should be injected for all types of apps

**Changes:** Make all kinds of app generate app.run.id

**API changes:** None

**Upgrade Instructions:** None

**Usage Instructions:** None

### CODE REVIEW

**Comment by Sanil15** [comment]
@bharathkk @lakshmi-manasa-g @prateekm please have a look

File: samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java [hunk 1]
```diff
@@ -127,9 +134,6 @@ final void writePlanJsonFile(String planJson) {
 
   private Map<String, String> getGeneratedConfig(String runId) {
```
Comment (lakshmi-manasa-g) [inline samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java:135]: does this method still need a runId passed into it?

Comment (Sanil15) [reply samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java]: nice catch, cleaned

**Comment by Sanil15** [comment]
> does it make sense to have a few unit tests to ensure run.id will be added to configs and also to ensure no regressions happen to this?

Added

**Pull request merged**